### ISSUE TITLE
Access feedback - Design intent Collaboration [MVP VA Mobile App]

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#14463]
**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Must
* Service Verification Letter PDF must be as accessible as we can make it. The end point will need to be reviewed for accessibility by the VA document remediation service. VSP a11y specialist can get you the link.
* * Veterans' Crisis Line modal (overlay) must have a way to be dismissed without scrolling to a cancel button at the bottom.

## Should
* Veterans' Crisis Line modal (overlay) should have a close button, prominently displayed.

## Consider
* https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432345660 Current step could have a secondary style like an extra (second) ring around it. Right now we're relying a bit too heavily on color. There's differences, but they're subtle. The gray (disabled) text might not pass a color contrast check right now, so will be something to keep an eye on.
* Service Verification Letter
    * There's a backend process that might take a few minutes, so I think it's worth exploring a confirmation screen letting the user know their letter is being produced.
    * Might also be worth looking at a push notification when the letter is ready
* < Back link text is slightly larger than the page heading. Is this intentional? Is there an opportunity to increase one or decrease the other to balance the hierarchy?

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

**Comment by bridgethapner** [comment]
Hi @1Copenut -- Thank you for the feedback! I don't understand the second item under Must re the [Veterans Crisis Line](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432348084). Users can exit the screen in the upper left corner and it will be its own screen (accessible from in-page CTAs), not a modal. Can you clarify the feedback here? Thank you!

**Comment by bridgethapner** [comment]
Hi there @kreek - Just wanted to flag the first item under Must: "Service Verification Letter PDF must be as accessible as we can make it. The end point will need to be reviewed for accessibility by the VA document remediation service." Is this something we can account for in the API work?

**Comment by kreek** [comment]
@bridgethapner 
> "Service Verification Letter PDF must be as accessible as we can make it. The end point will need to be reviewed for accessibility by the VA document remediation service." Is this something we can account for in the API work?

The PDF is generated by the EVSS service and is the same one returned to va.gov so maybe it has been reviewed already? Mobile display might change things. For things like tab order inside the PDF for screen readers that would have to change in EVSS.

**Comment by bridgethapner** [comment]
Got it, thank you @kreek! This is super helpful. VA.gov doesn't display the PDF (it just goes to downloaded files) so I'm guessing any API work to make the PDF more accessible in EVSS isn't done...is there a way to check the API to see if they do have any accessibility built-in? Once we know that we can develop an MVP solution for doing the best we can in front-end.

**Comment by bridgethapner** [comment]
Re our conversation earlier on PDF accessibility @bessgreen - I did a bit of digging and it doesn't look like iOS and Android can reliably read PDFs with their native voiceover functionality: "no fully accessible solutions are currently available to users who are blind with iOS’s VoiceOver and Android’s TalkBack screen readers." - https://www.levelaccess.com/pdf-accessibility-on-mobile-devices/

It also seems like the format of the PDF (VA seal, VA information, user name and address, letter content...) might make it difficult to read without restructuring in the API which is likely not in scope for MVP.

My gut is to fall back on fairly generic alt-text, but eager to hear if Brenda or Esther have a different POV.

**Comment by bessgreen** [comment]
@bridgethapner This is helpful context, thank you! I touched base with Esther & Brenda, and the short of it is that we agree with you to use the generic alt-text for each letter. Here's some context on why:

When considering whether or not the .pdf should be fully-readable by screen-readers, the initial thought might be that since this is text content, it should abide by the same rules as the text on the rest of the page. However, this [WCAG requirement](https://www.w3.org/TR/WCAG/#cc5) seems to allow for an exception of "web technologies" like PDFs. Whether PDFs can even be read by a screen reader when it's in the view screen will depend on the PDF viewer that's used. From a technical standpoint, we'd have to do some digging here to see if it's even feasible, and then investigate what's involved from a scope standpoint (not to mention any potential implications w/ the API that you mentioned above). 

Another element to consider is the broader use case for this feature. As we understand it, the primary use case for the Letters feature is to give users a copy of their letter(s) as "evidence" for an out-of-app need (e.g. providing proof of service to an employer). As a result, a user might not even need or want to read these letters from start-to-finish in order to complete their task. The primary function of this feature is for the user to be able to save the letter to their device's files - not to read it start-to-finish in the viewer.

TL;DR: the generic alt-text for these letters seems like the way to go for MVP. If it's critical that each letter must be read start-to-finish by a screen reader, this could be a good candidate to revisit in a future release.

**Comment by bridgethapner** [comment]
@bessgreen Thank you so much, this is super helpful to support this MVP decision. Per Trevor's guidance ("Service Verification Letter PDF must be as accessible as we can make it") I think we are ok here for November MVP to include alt text that would not necessarily read the generated PDF letters but be as detailed as we can be without exact voice over functionality. 

To improve the accessibility here, we'll need to have the endpoint reviewed for accessibility by the VA document remediation service to ensure we can make the PDFs more accessible soon after MVP. I'll create a ticket for this & follow up with VSP a11y specialists to start this process.  

cc @srmyers17 for awareness

**Comment by bridgethapner** [comment]
Notes for Trevor about a11y feedback: 

## Must

**Service Verification Letter PDF must be as accessible as we can make it. The endpoint will need to be reviewed for accessibility by the VA document remediation service. VSP a11y specialist can get you the link.** 

Since our mobile display (showing the PDF in-app) is different from than desktop display (which does not display the PDF), we are assuming the accessibility of the endpoint is not ready to support our design. To that end, I have:

1. Reached out in #vetsgov-accessibility to inquire after beginning the endpoint review process
2. Created #14731 for the work to review the endpoint for accessibility

For now, for MVP, our plan is to write alt-text that gets as detailed as possible, likely include the letter's description and context that the letter can be downloaded. For context, our MVP is launching to internal VA stakeholders only and we intend to improve the accessibility of letter PDFs before launching to the app stores.

**Veterans' Crisis Line modal (overlay) must have a way to be dismissed without scrolling to a cancel button at the bottom.** 

There is a "< Back" button to exit the screen in the top navigation bar.

## Should

**Veterans' Crisis Line modal (overlay) should have a close button, prominently displayed.**

There is a "< Back" button to exit the screen in the top navigation bar.

## Consider
**[Current step](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432345660) could have a secondary style like an extra (second) ring around it. Right now we're relying a bit too heavily on color. There's differences, but they're subtle. The gray (disabled) text might not pass a color contrast check right now, so will be something to keep an eye on.**

I added a 2pt dark blue outline to the active step ([wireframe](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432345660))

**Service Verification Letter: There's a backend process that might take a few minutes, so I think it's worth exploring a confirmation screen letting the user know their letter is being produced. Might also be worth looking at a push notification when the letter is ready.** 

I added [a loading screen](https://adhoc.invisionapp.com/share/GTZ1ESFF6BN#/screens/434431466_Letters_-Loading-) to let users know that a letter is being retrieved and the process might take a while. This is our MVP plan, and we may add a push notification later depending on how long letter retrieval takes. 

**< Back link text is slightly larger than the page heading. Is this intentional? Is there an opportunity to increase one or decrease the other to balance the hierarchy?**

We have standardized the header text to be 20px. This isn't reflected in the prototype but is the approach per conversations with out front-end developer.
